lib/trello/card.rb

```diff
@@ -516,14 +518,26 @@ def parse_writable_fields(fields, key)
     end
 
     def parse_readonly_fields(fields, key)
+      fields ||= {}
       gem_version_key = key.to_sym
       api_version_key = SYMBOL_TO_STRING[gem_version_key]
-      if fields.key?(api_version_key)
-        fields[api_version_key]
-      else
-        attributes[gem_version_key]
-      end
+
+      fields[api_version_key] || attributes[gem_version_key]
     end
 
+    def serialize_time(time)
+      return time unless time.is_a?(String)
```

Comment:
What could `time` be in this case? Should we return `time` if it is of a DateTime object?

Reply:
After some research, I think we should only support Time, DateTime, String only.

Should we write defensive code to check the object's class like this?

```ruby
return nil unless time.is_a?(String) || time.is_a?(Time) || time.is_a?(DateTime)
# or
raise ArgumentError unless time.is_a?(String) || time.is_a?(Time) || time.is_a?(DateTime)
```
Or just keep current feature only check the string?

### URL encode format difference between Time and DateTime:
Rest-client use `URI.encode_www_form_component` to format params.

```bash
t1 =Time.new(2019, 1, 1, 1, 1, 1, '+00:00')
t2 =DateTime.new(2019,1,1,1,1,1, '+00:00')

URI.encode_www_form_component(t1) #=> 2019-01-01+01%3A01%3A01+%2B0000
URI.encode_www_form_component(t2) #=> 2019-01-01T01%3A01%3A01%2B00%3A00
```

I also test both of this formats with Trello API. All works fine. So I think we can leave this difference here. In future, it's better to pre-format this field to a unified format before submit the request.

Reply:
> I also test both of this formats with Trello API. All works fine. So I think we can leave this difference here. In future, it's better to pre-format this field to a unified format before submit the request.

:+1: